imap_processing/utils.py

```diff
@@ -74,30 +88,38 @@ def convert_raw_to_eu(dataset: xr.Dataset, conversion_table_path, packet_name):
     """
     # Make sure there is column called "index" with unique
     # value such as 0, 1, 2, 3, ...
-    eu_conversion_table = pd.read_csv(
+    eu_conversion_df = pd.read_csv(
         conversion_table_path,
-        index_col="index",
+        **read_csv_kwargs,
     )
 
     # Look up all metadata fields for the packet name
-    metadata_list = eu_conversion_table.loc[
-        eu_conversion_table["packetName"] == packet_name
-    ]
+    packet_df = eu_conversion_df.loc[eu_conversion_df["packetName"] == packet_name]
 
     # for each metadata field, convert raw value to engineering unit
-    for field in metadata_list.index:
-        metadata_field = metadata_list.loc[field]["mnemonic"]
-        # On this line, we are getting the coefficients from the
-        # table and then reverse them because the np.polyval is
-        # expecting coefficient in descending order
-        coeff_values = metadata_list.loc[
-            metadata_list["mnemonic"] == metadata_field
-        ].values[0][6:][::-1]
-
-        # Convert the raw value to engineering unit
-        dataset[metadata_field].data = np.polyval(
-            coeff_values, dataset[metadata_field].data
-        )
+    for _, row in packet_df.iterrows():
+        if row["convertAs"] == "UNSEGMENTED_POLY":
+            # On this line, we are getting the coefficients from the
+            # table and then reverse them because the np.polyval is
+            # expecting coefficient in descending order
+            # coeff columns must have names 'c0', 'c1', 'c2', ...
+            coeff_values = row.filter(regex=r"c\d").values[::-1]
+
+            try:
+                # Convert the raw value to engineering unit
+                dataset[row["mnemonic"]].data = np.polyval(
+                    coeff_values, dataset[row["mnemonic"]].data
+                )
+                # Modify units attribute
+                if "unit" in row:
+                    dataset[row["mnemonic"]].attrs.update({"units": row["unit"]})
+            except KeyError:
```

Comment:
Ultimately, I think this should perhaps raise an exception. For now, the test data that I have does not match the updated packet definition which means that there is a conversion entry in the csv that is not present in the tests data... Growing pains